backend/routes/db.py

```diff
@@ -343,16 +275,164 @@ async def get_concepts_post_route(request: Request, id: Union[List[str], None] =
     return await get_concepts_route(request, id=id, table=table)
 
 
+sessions_cache: Dict = {}
+session_stale_hours = 24
+
+
+@router.get("/start-session")
+def start_session():
+    return get_session(None)
+
+
+def get_session(session_id: int):
+    if session_id:
+        if session_id not in sessions_cache:
+            # raise HTTPException(status_code=404, detail="Session not found")
+            raise Exception(f"Session {session_id} not found")
+        return sessions_cache[session_id]
+    session = {
+        "session_id": next_api_call_group_id(),
+        "session_start": datetime.now(),
+        "concept_ids_seen": [],
+        "searches": {},
+    }
+    return session
+
+
+def update_session(session_id: int, concepts: List[Dict]):
+    session = get_session(session_id)
+    session["concept_ids"].extend([c["concept_id"] for c in concepts])
+    session["last_used"] = datetime.now()
+    sessions_cache[session_id] = session
+    clear_old_sessions()
+    return session
+
+
+@throttle(wait = 600) # 10 minutes
+def clear_old_sessions():
+    for session_id in list(sessions_cache.keys()):
+        if datetime.now() - sessions_cache[session_id]["last_used"] > timedelta(hours=session_stale_hours):
+            del sessions_cache[session_id]
+    pdump(sessions_cache)
+
+
 @router.get("/concept-search")
-async def get_concepts_route(search: str) -> List:
-    q = """
-      SELECT *
+async def _concept_search(search_str: str, sort_by: str = "-total_cnt|vocabulary_id|concept_name") -> List[int]:
+    sort_columns = {"total_cnt", "concept_name", "vocabulary_id", "domain_id", "concept_class_id"}
+    #   for desc, prefix with -
+    sort_strs = sort_by.split('|')
+    sort_cols = []
+    for s in sort_strs:
+        desc = False
+        if s[0] == '-':
+            desc = True
+            s = s[1:]
+        if s not in sort_columns:
+            raise f"invalid sort_by: {s}"
+        s = f"{s} DESC" if desc else s
+        sort_cols.append(s)
+    q = f"""
+      SELECT concept_id
       FROM concepts_with_counts
-      WHERE concept_name ILIKE %s
+      WHERE concept_name ILIKE :search_str
```

Comment:
## Tri-grams?
I see `ILIKE`, but I don't see anything about that tri-gram extension here. Are we ditching that? If so, why?